rust/src/coordinator.rs

```diff
@@ -230,31 +227,24 @@ impl Coordinator {
         .ok_or(PetError::InvalidMessage)
     }
 
-    /// Freeze the seed dictionary. Fails due to insufficient update participants.
-    fn freeze_dict_seed(&mut self) -> Result<(), PetError> {
-        (self
-            .dict_seed
-            .values()
-            .all(|dict| dict.len() >= self.min_update))
-        .then_some(())
-        .ok_or(PetError::InsufficientParticipants)
-    }
-
     /// Update the mask dictionary.
-    fn update_dict_mask(&mut self, mask_url: &[u8]) {
-        self.dict_mask += mask_url
-            .chunks_exact(mask_url.len())
-            .map(|mask| mask.to_vec());
+    fn update_dict_mask(&mut self, mask_hash: &[u8]) {
+        self.dict_mask.update(iter::once(mask_hash.to_vec()))
```

Comment:
what's the reason for the change from `+=` to `.update()`? they are equivalent for `Counter<_>`, but the `+=` better conveys the semantics of counting, imo.

Reply:
Having a `+` operator with operands of that of different types was very confusing, especially with types that usually don't work with `+`: `Dict<K,V> + `Iterator<Item=V>` doesn't make much sense I think.

Reply:
mhh ok, that's probably quite subjective. i'm ok going with `.update()`.